### ISSUE TITLE
HttpClientTestingBackend.verify() drops query parameter info of open requests

### ISSUE THREAD

**Issue by stephanme** [angular/angular#22112]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior

On test failures, HttpClientTestingBackend.verify() prints out the list of open, unexpected requests as urls.
E.g. 
```
Error: Expected no open requests, found 1: GET /b/api/reportservice/uptimereport/monitors/1
    at HttpClientTestingBackend.verify (webpack:///C:/Users/***/node_modules/@angular/common/esm5/http/testing.js?:494:19)
```
Unfortunately, it drops the query parameters of the requested urls. This makes it more difficult to track down why a test failed.

## Expected behavior

E.g.
```
Error: Expected no open requests, found 1: GET /b/api/reportservice/uptimereport/monitors/1?startTime=-100&endTime=0
    at HttpClientTestingBackend.verify (webpack:///C:/Users/***/node_modules/@angular/common/esm5/http/testing.js?:494:19)
```

## Minimal reproduction of the problem with instructions

I haven't yet created a test to reproduce the problem but the questionable code seems to be at https://github.com/angular/angular/blob/5.2.0/packages/common/http/testing/src/backend.ts#L127

```
const url = testReq.request.urlWithParams.split('?')[0];
```

The split() call explicitly drops query params.

## What is the motivation / use case for changing the behavior?

Hard to track down failed tests for applications that use query parameters in backend requests.

## Environment

<pre><code>
Angular version: 5.2.0

Browser: does not apply
- [ ] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
```
ng --version

Angular CLI: 1.7.0-beta.1
Node: 9.2.0
OS: win32 x64
Angular: 5.2.0
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

@angular/cli: 1.7.0-beta.1
@angular-devkit/build-optimizer: 0.0.41
@angular-devkit/core: 0.0.28
@angular-devkit/schematics: 0.0.51
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.10.0-beta.1
@schematics/angular: 0.1.16
typescript: 2.6.2
webpack: 3.10.0
```
Others:

</code></pre>

**Comment by textbook** [comment]
I think this is related to #19974

**Comment by JoostK** [comment]
This was addressed in #44917.
